### PR TITLE
Retry logic for `processRecords`

### PR DESCRIPTION
**Pull request by dubiousdavid**
-   Added retry logic to `processRecords`. Downstream libraries should only throw if
    the error is transient and can be retried with eventual success.
-  Changed to ES module since we're relying on `p-retry` now.

### CODE REVIEW

File: src/mongoChangeStream.test.ts [hunk 1]
```diff
@@ -127,36 +127,36 @@ const initState = async (sync: SyncObj, db: Db, coll: Collection) => {
 }
 
 describe('syncing', () => {
-  test('starting change stream is idempotent', async () => {
+  // NOTE: This test is flaky
```
Comment (daveyarwood) [inline src/mongoChangeStream.test.ts:130]: Just curious, what's flaky about it? I don't see any assertions. Does it sometimes throw an error?

File: src/mongoChangeStream.test.ts [hunk 2]
```diff
@@ -224,6 +224,37 @@ describe('syncing', () => {
...
+      }
+      processed.push(...docs)
+    }
+    const batchSize = 100
+    const initialScan = await sync.runInitialScan(processRecords, { batchSize })
+    // Wait for initial scan to complete
+    await initialScan.start()
+    assert.equal(processed.length, numDocs)
+    // Expect one extra call to processRecords
+    assert.equal(counter, numDocs / batchSize + 1)
```
Comment (daveyarwood) [inline src/mongoChangeStream.test.ts:253]: Nice assertion!

File: src/mongoChangeStream.test.ts (hunk 1 above)
Comment (dubiousdavid) [reply src/mongoChangeStream.test.ts:130]: I don't fully understand why after spending around 30 minutes looking into it.

File: src/mongoChangeStream.test.ts (hunk 2 above)
Comment (dubiousdavid) [reply src/mongoChangeStream.test.ts:253]: Thanks!

**Review by daveyarwood — APPROVED** [review]
Looks great!

I think now, we could potentially remove p-retry from mongo2crate and mongo2elastic and let mongochangestream handle the retry logic.

**Pull request merged**